code/game/objects/structures/plasticflaps.dm

```diff
@@ -47,13 +58,13 @@
 		return TRUE
 	set_anchored(!anchored)
 	update_atmos_behaviour()
-	air_update_turf(TRUE)
 	to_chat(user, span_notice("You [uraction] the floor."))
 	return TRUE
 
 ///Update the flaps behaviour to gases, if not anchored will let air pass through
 /obj/structure/plasticflaps/proc/update_atmos_behaviour()
-	can_atmos_pass = anchored ? ATMOS_PASS_YES : ATMOS_PASS_NO
+	can_atmos_pass = anchored ? ATMOS_PASS_NO : ATMOS_PASS_YES
+	air_update_turf(TRUE, FALSE)
```

Comment:
FALSE should only be used if we are ADDING a new way for turfs to connect
If we're adding a block, use TRUE.

If you don't, you can cause excited groups to stay connected despite being improperly formed

Reply:
What vars are used need to be conditional based off what you actually want to do